src/components/views/rooms/BasicMessageComposer.tsx

```diff
@@ -630,6 +635,14 @@ export default class BasicMessageEditor extends React.Component<IProps, IState>
         this.setState({ completionIndex });
     };
 
+    private configureUseMarkdown = (): void => {
+        const useMarkdown = SettingsStore.getValue("MessageComposerInput.useMarkdown");
+        this.setState({ useMarkdown });
+        if (!useMarkdown && this.formatBarRef.current) {
+            this.formatBarRef.current.hide();
+        }
```

Comment:
This probably would benefit from being moved to `componentDidUpdate` and checking that the `useMardown` state value has changed